exporter/opencensusexporter/opencensus.go

```diff
@@ -90,6 +90,10 @@ func createOCAgentExporter(logger *zap.Logger, config configmodels.Exporter, opt
 
 	exportersChan := make(chan *ocagent.Exporter, numWorkers)
 	for exporterIndex := 0; exporterIndex < numWorkers; exporterIndex++ {
+		// TODO: ocagent.NewExporter blocks for connection. Now that we have ability
```

Comment:
There is a bit of nuance in this regard: for intermittent network issues we don't want the ocagent to report fatal error, it should just re-establish the connection in the background.

Reply:
Looking at ocagent.Exporter code, it doesn't report errors when it cannot connect, it does re-connections, so it already works as you expected. It may report an error in other situations but it blocks until first connection attempt is either successful or failed which is not good, it blocks the startup of the whole collector.
In my sample runs OpenCensus exporters takes about 3-4 seconds of startup time (when it connects successfully), which is undesirable.
The new async Host communication method is a good solution to this problem.

Reply:
Yeah, it is a good improvement. I'm just wondering about the cases in which an exporter should report fatal error.